### PR TITLE
[GITHUB-7] Update to Latest Standards

### PR DESCRIPTION
**Pull request by dennisconrad**
* Flatten variables
* Convert to molecule testing
* Use sansible.users_and_groups
* "Bugfixes and performance improvements"

### CODE REVIEW

File: meta/main.yml [hunk 1]
```diff
@@ -1,16 +1,35 @@
...
       versions:
-        - precise
         - trusty
+        - xenial
+        - bionic
   categories:
     - development
+
+dependencies:
+  - role: sansible.users_and_groups
```
Comment (lobsterdore) [inline meta/main.yml:17]: I think you can just have just one of these and assert that both user and group are defined, they are intended to be used together:

```YAML
  - role: sansible.users_and_groups
    sansible_users_and_groups_groups:
      - name: "{{ sansible_ssmtp_group }}"
    sansible_users_and_groups_users:
      - name: "{{ sansible_ssmtp_user }}"
        group: "{{ sansible_ssmtp_group }}"
    when:
        - sansible_ssmtp_user is not none
        - sansible_ssmtp_group is not none
```

File: tasks/install.yml [hunk 2]
```diff
@@ -1,31 +1,48 @@
...
-- name: Add ssmtp group for restricted usage
-  become: yes
-  group:
     name: ssmtp
-    state: present
-  when: ssmtp.restrict
 
-- name: Restrict ssmtp usage
-  become : yes
+- name: "Restrict SSMTP Usage to User {{ sansible_ssmtp_user }}"
```
Comment (lobsterdore) [inline tasks/install.yml]: Again I think we just need one of these and assert that both user and group are specified.

File: meta/main.yml (hunk 1 above)
Comment (dennisconrad) [reply meta/main.yml:17]: Added an assert to the tasks.

**Comment by lobsterdore** [comment]
LGTM 👍

**Review by rsgoodall — APPROVED** [review]

**Pull request merged**